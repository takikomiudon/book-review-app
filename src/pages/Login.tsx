import React, { MouseEvent, useState } from "react";

const Login = () => {
  const [email, setEmail] = useState("");
  const [password, setPassword] = useState("");

  const handleSubmit = (e: MouseEvent<HTMLButtonElement>) => {
    e.preventDefault();
  }

  return (
    <div>
      <form>
        <label>メールアドレス</label>
        <br />
        <input
          type="email"
          value={email}
          onChange={(e) => setEmail(e.target.value)}
        />
        <br />
        <label>パスワード</label>
        <br />
        <input
          type="password"
          value={password}
          onChange={(e) => setPassword(e.target.value)}
        />
        <br />
        <button type="submit" onClick={handleSubmit}>
          Submit
        </button>
      </form>
    </div>
  );
};

export default Login;
