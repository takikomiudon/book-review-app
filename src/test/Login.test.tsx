import React from 'react';
import { render } from '@testing-library/react';
import Login from '../pages/Login';
import { expect } from '@jest/globals';

test("exist components", () => {
  const view = render(<Login />);
  expect(view.container).toMatchSnapshot();
});