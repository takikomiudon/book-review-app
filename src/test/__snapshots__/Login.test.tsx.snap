// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`exist components 1`] = `
<div>
  <div>
    <form>
      <label>
        メールアドレス
      </label>
      <br />
      <input
        type="email"
        value=""
      />
      <br />
      <label>
        パスワード
      </label>
      <br />
      <input
        type="password"
        value=""
      />
      <br />
      <button
        type="submit"
      >
        Submit
      </button>
    </form>
  </div>
</div>
`;
